### PR TITLE
feature: add CI and test cases for ngx lua.

### PR DESCRIPTION
**Pull request by moonming**

### CODE REVIEW

**Review by wu-sheng — COMMENTED** [review]
Discussion about pure LUA and Nginx native LUA happens in the mail list.

https://lists.apache.org/thread.html/r24aa5e7673db7132352122e76f1ac398c80c5812fa9864d496b8a581%40%3Cdev.skywalking.apache.org%3E

Waiting for the result from there.

File: lib/skywalking/segment_ref.lua [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+--
 -- Unless required by applicable law or agreed to in writing, software
 -- distributed under the License is distributed on an "AS IS" BASIS,
 -- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
--- 
-
+--
+local ngx_re = require("ngx.re")
```
Comment (wu-sheng) [inline lib/skywalking/segment_ref.lua]: Do you miss the change here? `segment_ref` is a core related part.

Comment (wu-sheng) [reply lib/skywalking/segment_ref.lua]: I have read there is a compatible in util.lua.

File: lib/skywalking/util.lua [hunk 2]
```diff
@@ -1,34 +1,92 @@
...
+end
+
+_M.split = split
+_M.timestamp = timestamp
 
-local Util = {}
 local MAX_ID_PART2 = 1000000000
 local MAX_ID_PART3 = 100000
-local SEQ = 1
+local metadata_buffer = ngx.shared.tracing_buffer
```
Comment (wu-sheng) [inline lib/skywalking/util.lua]: I think this is incompatible for pure lua.

File: lib/skywalking/util.lua [hunk 3]
```diff
@@ -1,34 +1,92 @@
...
+    if not seed then
+        seed = ngx.now() * 1000 + ngx.worker.pid()
+    end
+
+    return seed
+end
+
+math.randomseed(random_seed())
+
+function _M.newID()
```
Comment (wu-sheng) [inline lib/skywalking/util.lua:85]: The ID algorithm should be good enough even without seq. `ngx.shared.tracing_buffer` force them to increase strictly, which is not necessary. In other language, it just provided a static int to do `i++`. If this is hard in Lua or nginx lua case, feel free to delete this `seq`.

File: t/util.t [hunk 4]
```diff
@@ -0,0 +1,96 @@
+use Test::Nginx::Socket 'no_plan';
```
Comment (wu-sheng) [inline t/util.t:1]: What is this file for? Could you give me some guides?

File: lib/skywalking/segment_ref.lua (hunk 1 above)
Comment (moonming) [reply lib/skywalking/segment_ref.lua]: I forgot to fix here. I will add back.

File: lib/skywalking/util.lua (hunk 3 above)
Comment (moonming) [reply lib/skywalking/util.lua:85]: got it. `seq` is hard to implement in ngx lua, I will remove `seq`.

File: t/util.t (hunk 4 above)
Comment (moonming) [reply t/util.t:1]: It's test framework in Nginx lua: https://metacpan.org/pod/Test::Nginx::Socket, most of nginx lua module are test by `test::nginx`.

File: lib/skywalking/util.lua (hunk 2 above)
Comment (moonming) [reply lib/skywalking/util.lua]: I will remove it.

**Review by wu-sheng — APPROVED** [review]

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**